draft-ietf-tls-esni.md

```diff
@@ -804,7 +797,30 @@ connection in this case.
 
 # Security Considerations
 
-## Why is cleartext DNS OK? {#cleartext-dns}
+## Security and Privacy Goals
+
+Informally, the primary security and privacy goals of ECH against an active attacker are
+as follows:
+
+1. Use of ECH does not weaken the security properties of TLS without ECH.
+2. TLS connection establishment to a host with a specific ECHConfig is indistinguishable
+from a connection to any other host with the same ECHConfig. That is, given an anonymity
+set of servers of size k sharing the same ECHConfig, an attacker has negligible advantage
+in correctly identifying the host for a given connection with probability larger than 1/k.
```

Comment:
Same ECHConfig is necessary, but not sufficient I think. If the hosts share an ECHConfig but have not other TLS config (the deleted paragraph above), the hosts are distinguishable. (I've lost track of whether the current level of ClientHello binding is sufficient to limit this to anonymity within the negotiated values or the full config...) The third paragraph mentions this, but it seems to be clarifying point (2), but point (2) doesn't actually imply this.

Reply:
Fair :) I was trying to get away without too much text. I can clarify that this is necessary (and rework text accordingly).

Reply:
@davidben, reworked. Please have another look!